frappe/desk/doctype/event/event.py

```diff
@@ -392,6 +393,62 @@ def add_event(e, date):
 
 				remove_events.append(e)
 
+			if e.repeat_on == "Half Yearly":
```

Comment:
Add basic tests for these. 

There's "freeze_time" utility in test utils to evaluate code at some date.

Reply:
done